api/src/Validator/AllowTransition/AssertAllowTransitionsValidator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Validator\AllowTransition;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\InvalidArgumentException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class AssertAllowTransitionsValidator extends ConstraintValidator {
+    public function __construct(public RequestStack $requestStack) {
+    }
+
+    public function validate($value, Constraint $constraint) {
+        if (!$constraint instanceof AssertAllowTransitions) {
+            throw new UnexpectedTypeException($constraint, AssertAllowTransitions::class);
+        }
+
+        $transitions = $constraint->getTransitions();
+        if (!$transitions->forAll(fn ($_, $elem) => self::isValidTransition($elem))) {
+            throw new InvalidArgumentException('All transitions must have a from and a to key');
+        }
+
+        $transitions = $transitions->map(fn ($elem) => new Transition($elem['from'], $elem['to']));
+
+        $allFrom = $transitions->map(fn (Transition $elem) => $elem->from)->toArray();
+        $allTo = $transitions->map(fn (Transition $elem) => $elem->to)->toArray();
+        $allTo = array_merge(...$allTo);
+        if (!(new ArrayCollection($allTo))->forAll(fn ($_, $to) => in_array($to, $allFrom, true))) {
+            throw new InvalidArgumentException('All to must appear in a from again');
+        }
+
+        $previousObject = $this->requestStack->getCurrentRequest()->attributes->get('previous_data');
+        $previousValue = $previousObject->{$this->context->getPropertyName()};
+
+        if ($value === $previousValue) {
```

Comment:
Wir unterstützen es aktuell also nicht, dass der Wert zwingend geändert werden muss, korrekt? Du hast auch einen Test dafür, daher für mich okay. Wenn wirs mal brauchen können wir das auch via ein Flag auf der Validierung aktivierbar machen.

Reply:
Es wird immer die ganze Entity validiert. Aber beim lesen deiner Validierungs-Annotation habe ich mich gefragt ob ich auch "from invited to invited" etc. erlauben müsste. Darum dieser Kommentar

Reply:
Ah, ob man es explizit erlauben müsste. Fände ich bisschen komisch, da immer die ganze Entity validiert wird. das so ein common pitfall, den ich vermeiden würde.